Detectors/EMCAL/simulation/src/DMAOutputStream.cxx

```diff
@@ -41,30 +53,30 @@ void DMAOutputStream::writeData(RawHeader header, gsl::span<char> buffer)
   // from the maximum possible payload size
   constexpr int MAXNWORDS = PAGESIZE - sizeof(header) - sizeof(uint32_t);
   bool writeNext = true;
-  int pagecounter = 0, currentindex = 0;
+  int pagecounter = header.pageCnt, currentindex = 0;
   while (writeNext) {
     int sizeRemain = buffer.size() - currentindex;
     int nwordspage = MAXNWORDS;
     if (sizeRemain < MAXNWORDS) {
       // Last page
       nwordspage = sizeRemain;
       writeNext = false;
-      header.stop = true;
     }
-    header.packetCounter = pagecounter;
+    header.pageCnt = pagecounter;
     header.memorySize = nwordspage + sizeof(RawHeader);
     header.offsetToNext = 8192;
 
     writeDMAPage(header, gsl::span(buffer.data() + currentindex, nwordspage), PAGESIZE);
 
     if (writeNext) {
       currentindex += nwordspage;
-      pagecounter++;
     }
+    pagecounter++;
   }
+  return pagecounter;
 }
 
-void DMAOutputStream::writeDMAPage(const RawHeader& header, gsl::span<char> payload, int pagesize)
+void DMAOutputStream::writeDMAPage(const RawHeader& header, gsl::span<const char> payload, int pagesize)
 {
   std::vector<char> dmapage(pagesize);
   o2::header::RAWDataHeaderV4* outheader = reinterpret_cast<o2::header::RAWDataHeaderV4*>(dmapage.data());
```

Comment:
to faciliate later change to new header version, it is better to use RawHeader (which you have already aliased to RDHv4)

Reply:
Yes, will fix this. Eventually I will template the class as done for the raw reader.